### ISSUE TITLE
Thin Mailbox 

### ISSUE THREAD

**Issue by SchrodingerZhu** [tisonkun/tijkv#1]
## Problem

Currently, the implementation relies on global references. As @tisonkun has said, this will cause problems and become performance bottleneck as the number of global references are limited.

## Solution

Implement a thin communication layer between `rust` and` jvm` async runtime to enable JVM to dispatch the response from `rust` world, which will eliminate the usage of global references per request. The communicate queue in the JVM world can be set up as a static field, so that we do not even need to store any global reference for the client instance.

## Details

We may need to find a way to let JVM poll on a concurrent queue, for example https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentLinkedQueue.html. 

Basically, JVM can store a reference to `rust` object safely, holding the ownership in the same time. Therefore, sending messages from Java to Rust can be much easier.

There may be at least two variants of message modes to consider:
- One-shot 
- Stream

both are manageable with the thin communication layer.

We need, also, develop a way for JVM to identify the receivers.

**Comment by tisonkun** [comment]
I think the `AsyncBookKeeper` introduced in 1386a958114c198e398d858bde8c4b40888df9be simply implements real async between Java & Rust. But your proposal is a more formal abstraction.
